tests/neg/gadt-eval.scala

```diff
@@ -0,0 +1,33 @@
+sealed trait Exp[T]
+case class Lit(value: Int) extends Exp[Int]
+case class Pair[A, B](fst: Exp[A], snd: Exp[B]) extends Exp[(A, B)]
+
+object Test {
+  def eval[T](e: Exp[T]): T = e match {
+    case Lit(x) =>
+      x
+    case Pair(a, b) =>
+      (eval(a), eval(a))
+        // -- [E007] Type Mismatch Error: tests/neg/gadt-eval.scala:10:6 ------------------
+        // 10 |      (eval(a), eval(a))
+        // |      ^^^^^^^^^^^^^^^^^^
+        // |    found:    (_$1, _$1)
+        // |    required: T
+        // |
+        // |    where:    T is a type in method eval which is an alias of (_$1, _$2)
```

Comment:
This overall testcase is cool!

Wondering how to explain where these type variables come from. They come from the match (hard to show), and they show up in the types of a and b, which is easier to show, most easily by showing the whole typing context. That might be too much, but I think GHC sometimes does it, and both Agda and Coq absolutely do it and have to do it. Not sure where to draw a line.

Reply:
Yeah, if you can think of a good way to display this information I'm interested, I haven't checked what other compilers do.

Reply:
I don't get it. If "T is a type in method eval which is an alias of (_$1, _$2)", why is there a type error?

Reply:
```
found: (_$1, _$1) 
expected: (_$1, _$2) 
```
The second type parameter of the tuple differs